src/sortStudents.ts

```diff
@@ -1,16 +1,104 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function calculateAverageGrade(grades: number[]): number {
+  if (grades.length === 0) {
+    return 0;
+  }
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  const sum = grades.reduce((acc, grade) => acc + grade, 0);
+
+  return sum / grades.length;
+}
+
+function sortString(
+  array: [],
+  parameter: 'name' | 'surname',
+  order: SortOrder,
+): [] {
+  if (order === 'desc') {
+    return array.sort(
+      (a: Student, b: Student) => b[parameter].localeCompare(a[parameter]),
+    );
+  }
+
+  return array.sort(
+    (a: Student, b: Student) => a[parameter].localeCompare(b[parameter]),
+  );
+}
+
+export function sortStudents(
+  students: [],
+  sortBy: SortType,
+  order: SortOrder,
+)
+  :[] {
+  const result: [] = JSON.parse(JSON.stringify(students));
+
+  switch (sortBy) {
+    case SortType.Name:
+      return sortString(result, 'name', order);
+
+    case SortType.Surname:
+      return sortString(result, 'surname', order);
+
```

Comment:
we can group these cases
```suggestion
    case SortType.Name:
    case SortType.Surname:
      return sortString(result, sortBy, order);

```